src/peer_manager.rs

```diff
@@ -734,6 +734,26 @@ impl PeerManager {
             .map(|(name, _)| *name);
         for name in removal_keys.iter() {
             let _ = self.candidates.remove(name);
+            let _ = self.peer_map.remove_by_name(&name).and_then(|peer| match peer {
```

Comment:
if we chain these two, the log at line 701 becomes incorrect.
not sure if this will cause problem for our invariant check script